webpack/assets/javascripts/react_app/redux/reducers/factChart/index.js

```diff
@@ -0,0 +1,31 @@
+import Immutable from 'seamless-immutable';
+import {
+  FACT_CHART_DATA_SUCCESS,
+  CLOSE_MODAL,
+  OPEN_MODAL,
+} from '../../consts';
+
+const initialState = Immutable({
+  showModal: false,
+  chartData: [],
+  loaderStatus: 'PENDING',
+  title: '',
+});
+
+export default (state = initialState, action) => {
+  switch (action.type) {
+    case FACT_CHART_DATA_SUCCESS:
```

Comment:
where are the implementation for FACT_CHART_DATA_FAILURE and FACT_CHART_DATA_REQUEST?

Reply:
pie chart component displays spinner while waiting and an error message when there's no data.
what should we do in request and failure here?

Reply:
IMHO once the action are fired, than we should change the state, for example, on failure - set the correct title etc.
on loading, it should set the loaderStatus correctly etc.